core/src/main/java/org/elasticsearch/cluster/routing/allocation/IndexMetaDataUpdater.java

```diff
@@ -39,23 +44,23 @@
  * Observer that tracks changes made to RoutingNodes in order to update the primary terms and in-sync allocation ids in
  * {@link IndexMetaData} once the allocation round has completed.
  *
- * Primary terms are updated on primary initialization or primary promotion.
+ * Primary terms are updated on primary initialization or when an active primary fails.
  *
  * Allocation ids are added for shards that become active and removed for shards that stop being active.
  */
 public class IndexMetaDataUpdater extends RoutingChangesObserver.AbstractRoutingChangesObserver {
     private final Map<ShardId, Updates> shardChanges = new HashMap<>();
 
     @Override
-    public void shardInitialized(ShardRouting unassignedShard) {
-        if (unassignedShard.primary()) {
-            increasePrimaryTerm(unassignedShard);
-        }
-    }
+    public void shardInitialized(ShardRouting unassignedShard, ShardRouting initializingShard) {
```

Comment:
its not clear to me why we need both the `unassignedShard` and `initializingShard` - only the `initializingShard` is used here and I don't see the `unassignedShard` used in any of the other implementations of `RoutingChangesObserver`